File: src/main/java/no/fintlabs/validation/OnlyOneTimeFilterType.java
package no.fintlabs.validation;

import jakarta.validation.Constraint;
import jakarta.validation.Payload;

import java.lang.annotation.Retention;

import static java.lang.annotation.RetentionPolicy.RUNTIME;

@Retention(RUNTIME)
@Constraint(validatedBy = OnlyOneTimeFilterTypeValidator.class)
public @interface OnlyOneTimeFilterType {

    String message() default "contains multiple time filters";

    Class<?>[] groups() default {};

    Class<? extends Payload>[] payload() default {};

}

File: src/main/kotlin/no/novari/flyt/history/validation/OnlyOneTimeFilterType.kt
package no.novari.flyt.history.validation

import jakarta.validation.Constraint
import jakarta.validation.Payload
import kotlin.reflect.KClass

@Target(AnnotationTarget.CLASS)
@Retention(AnnotationRetention.RUNTIME)
@Constraint(validatedBy = [OnlyOneTimeFilterTypeValidator::class])
annotation class OnlyOneTimeFilterType(
    val message: String = "contains multiple time filters",
    val groups: Array<KClass<*>> = [],
    val payload: Array<KClass<out Payload>> = [],
)
